agents/matlab/matlab_agent/src/core/interactive.py

```diff
@@ -0,0 +1,302 @@
+        )
+
+        self.start_time: Optional[float] = None
+        self.sequence = 0
+
+    # ------------------------------------------------------------------
+    def _start_matlab(self) -> None:
+        cmd = [
+            "matlab",
+            "-batch",
+            f"addpath('{
+                self.sim_path}');cd('{
+                self.sim_path}');run('{
+                self.sim_file}');",
+        ]
+        self.out_srv.matlab_proc = subprocess.Popen(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+
+    def start(self, pm: PerformanceMonitor) -> None:
+        """Start the TCP servers and MATLAB process.
+        Args:
+            pm (PerformanceMonitor): Performance monitor instance.
+        """
+        self.start_time = time.time()
+        self.out_srv.start()
+        self.in_srv.start()
+        logger.debug("[INTERACTIVE] TCP servers started on %s:%s and %s:%s",
+                     self.out_srv.host, self.out_srv.port,
+                     self.in_srv.host, self.in_srv.port)
+        self._start_matlab()
+        logger.debug("[INTERACTIVE] Waiting for MATLAB to start...")
+        pm.record_matlab_startup_complete()
+        self.out_srv.accept()
+        self.in_srv.accept()
+        # Perform a handshake with MATLAB
+        self.out_srv.send({})  # handshake
+
+    # ------------------------------------------------------------------
+    def _relay(self, payload: Dict[str, Any]) -> None:
+        """Relay a response from the MATLAB process to the broker
+        including the full simulation output."""
+
+        # Costruiamo il messaggio base con l’utility già esistente
+        msg = create_response(
+            "interactive",
+            self.sim_file,
+            "interactive",
+            self.templates,
+            data=payload,
+            sequence=self.sequence,
+            bridge_meta=self.bridge_meta,
+            request_id=self.request_id,
+        )
+        msg["output"] = payload
+        self.broker.send_result(self.source, msg)
+        self.sequence += 1
+
+    @staticmethod
+    def _only_inputs(frame: Dict[str, Any]) -> Dict[str, Any]:
+        """Extract only the inputs from the frame."""
+        if isinstance(frame, dict):
+            sim = frame.get("simulation")
+            if isinstance(sim, dict) and "inputs" in sim:
+                logger.debug("[INTERACTIVE] Received inputs: %s", sim["inputs"])
+                return sim["inputs"]
+        return frame
+
+    def run(self, pm: PerformanceMonitor, msg_dict: Dict[str, Any]) -> None:
+        """Run the interactive simulation loop."""
+        sim = msg_dict["simulation"]
+        stream_key = sim["inputs"]["stream_source"].replace("rabbitmq://", "")
+
+        ch = self.broker.channel
+        qname = f"Q.{self.agent_id}.interactive.{self.request_id}"
+        ch.exchange_declare(
+            "ex.input.stream",
```

Comment:
These fields already exist in `config.yaml`. Can they be taken from there? If not, please move these to either `util/settings.py` or `util/constants.py`.